av/stream.pyx

```diff
@@ -87,17 +87,23 @@ cdef class Stream(object):
         if self.container.ptr.iformat:
 
             # Find the codec.
-            self._codec = lib.avcodec_find_decoder(self._codec_context.codec_id)
+            h264_id = lib.avcodec_find_decoder_by_name('h264').id
+            h265_id = lib.avcodec_find_decoder_by_name('hevc').id
+            my_id = self._codec_context.codec_id
+            if container.hwaccel and my_id == h264_id:
+                self._codec = lib.avcodec_find_decoder_by_name('h264_cuvid')
+            elif container.hwaccel and my_id == h265_id:
+                self._codec = lib.avcodec_find_decoder_by_name('hevc_cuvid')
+            else:
+                self._codec = lib.avcodec_find_decoder(self._codec_context.codec_id)
```

Comment:
We need to treat any codec with NVIDIA acceleration as special case ‒ FFMPEG won't choose codec from external library as default one (as stated in comment in FFMPEG's allcodecs.c). I added comment to mention that.

I chose h264 and h265, because those are the ones that we are currently using.